File: Packets/ProtocolVersion.vb
Public Class ProtocolVersionPacket
    Inherits Packet
    Public Overrides Property OPCode As Byte = &H0
    Public Const Length As Byte = &H8
    Public Const Version As UInt32 = 636

    Sub New()
        MyBase.New()
    End Sub

    Public Overrides Function GetByteArray() As Byte()
        Dim p As New PacketBuilder
        p.Write(Version)
        Payload = p.GetBytes
        Return Package(False, True)
    End Function
End Class

File: Packets/ProtocolVersion.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
public class ProtocolVersionPacket : Packet
{
	public override byte OPCode { get; set; }
	public const byte Length = 0x8;

	public const UInt32 Version = 642;
	public ProtocolVersionPacket() : base()
	{
	}

	public override byte[] GetByteArray()
	{
        OPCode = 0x0;
		PacketBuilder p = new PacketBuilder();
		p.Write(Version);
		Payload = p.GetBytes();
		return Package();
	}
}
